File: Day-3/part1.pl
#!perl -pF
push@s,[@F]}{map{$_=join'',@{$s[$y=$_]};s^\d+^$p=0;$%=$&;for$x($-[0]..$+[0]-1){for$v(-1..1){$p|=$%*(0<=$y+$v<140&0<=$x+$_<140&$s[$y+$v][$x+$_]!~/[\d.]/)for-1..1}}$t+=$p^eg}0..139;$_=$t

File: Day-14/part1.rb
#!ruby
a = $<.map{_1.chomp.chars}.transpose

a.map!{|c|
    s = -1
    while s != 0
        s = 0
        for i in 1...c.size
            if c[i-1] == ?. && c[i] == ?O
                c[i-1], c[i] = c[i], c[i-1]
                s += 1
            end
        end
    end

    c
}

a=a.transpose.map(&:join)
p a.size.times.sum{|i| a[i].count(?O) * (a.size - i)}